include/rapidcheck/detail/Unicode.hpp

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
+namespace rc {
+namespace detail {
+
+template<typename T, typename RandomType>
```

Comment:
The reason it is templated on T is that it seemed to be the existing convention. Though I guess it is fairly unnecessary when it just generated an integer type, those tend to be fairly each to assign to each other. So yes, this function probably would be better off as a non-template.